src/transform/plugins/links/index.ts

```diff
@@ -206,28 +206,23 @@ function processLink(
             tokens,
             idx,
             nextToken,
-            href,
+            href: originalHref,
             currentPath,
             log,
             cache,
         });
     }
 
-    let newPathname = '';
+    const patchedHref =
+        !isAbsolute(originalHref) && !originalHref.includes('//')
+            ? url.format({
+                  ...url.parse(originalHref),
+                  pathname: getPublicPath(opts, file),
+              })
+            : originalHref;
+    const linkHrefTransformer = transformLink || defaultTransformLink;
 
-    if (!isAbsolute(href) && !href.includes('//')) {
-        newPathname = getPublicPath(opts, file);
-
-        href = url.format({
-            ...url.parse(href),
-            pathname: newPathname,
-        });
-    }
-
-    if (pathname || newPathname) {
-        const transformer = transformLink || defaultTransformLink;
-        linkToken.attrSet('href', transformer(href));
-    }
+    linkToken.attrSet('href', linkHrefTransformer(patchedHref));
```

Comment:
There used to be a check for `if (pathname || newPathname)` here. Is it removed intentionally?